pkg/tenant/db/db_test.go

```diff
@@ -1,21 +1,306 @@
 package db_test
 
 import (
+	"bytes"
+	"context"
+	"fmt"
 	"testing"
 
 	"github.com/rotationalio/ensign/pkg/tenant/config"
 	"github.com/rotationalio/ensign/pkg/tenant/db"
 	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"github.com/trisacrypto/directory/pkg/trtl"
+	trtlconfig "github.com/trisacrypto/directory/pkg/trtl/config"
+	"github.com/trisacrypto/directory/pkg/trtl/mock"
+	pb "github.com/trisacrypto/directory/pkg/trtl/pb/v1"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
+//===========================================================================
+// Database Test Suite
+//===========================================================================
+
+type dbTestSuite struct {
+	suite.Suite
+	mock *mock.RemoteTrtl
+}
+
+func (s *dbTestSuite) SetupSuite() {
+	require := s.Require()
+	require.NoError(db.Connect(config.DatabaseConfig{Testing: true}), "unable to connect to db in testing mode")
+	require.True(db.IsTesting(), "expected database to be in testing mode")
+	require.True(db.IsConnected(), "expected database to be connected in testing mode")
+
+	// Add the mock to the suite for ease of use
+	s.mock = db.GetMock()
+}
+
+func (s *dbTestSuite) TearDownSuite() {
```

Comment:
TODO: Add `s.mock.Reset()` to a suite `After` function (make sure reset is called after every test).